vxwhatsapp/consumer.py

```diff
@@ -94,26 +94,27 @@ async def process_message(self, message: IncomingMessage):
             await message.reject(requeue=False)
             return
 
-        logger.debug(f"Processing outbound message {msg}")
-        try:
-            await self.submit_message(msg)
-        except aiohttp.ClientResponseError as e:
-            # If it's a retryable upstream error, requeue the message
-            if e.status > 499:
-                await message.reject(requeue=True)
-                # If we've retried this before, wait before retrying again
-                if message.redelivered:
-                    await sleep(0.5)
-            else:
-                # Otherwise log the error and reject
-                logger.exception(f"Upstream HTTP error processing {msg}")
+        async with message.process():
```

Comment:
The problem with `message.process` is that it does the `message.ack` or `message.reject` for us, which means we can't do it below, otherwise we get an error.

Usually this is fine, because if there's an exception raised, then it will reject, and if there's no exception, then it will ack. The problem is for retryable errors, we want to reject with requeue=True, and we can't do that with `message.process`, so we need to catch all exceptions ourselves, and make sure we're always calling `ack` or `reject`.

What we could do is have a retry for the timeout error that you're seeing, but it's probably better to either make sure that the upstream service can respond faster, or to increase the timeout so that we're never hitting it under normal circumstances.